### DISCUSSION TITLE
Join Field Drawer Behaviour

### DISCUSSION THREAD

**Discussion by slavanossar** [payloadcms/payload#9119] [Feature Requests & Ideas]
When creating a document using the `+` button on a `relationship` field, the drawer remains open after saving. For a `join` field, the drawer immediately closes after saving.

It would be great if the `join` fields had the same behaviour as the `relationship` field. It would be even better if both fields had an `admin.closeOnSave` option to control this behaviour, since either one could be desired.

My use case is that the document being created via the `join` field has it's own `join` field, which can't be edited until there is an `ID` available to pass to to the drawer. This is the current experience:

- Click the `join` field's `Create New` button
- Fill in all fields (not possible to add rows to `join` field yet)
- Click `Save`, triggering drawer to close
- Click the `Edit` button on the new row in the `join` field table to open the drawer again
- Add items to the nested `join` field

**Comment by DanRibbens** [comment]
This all makes sense to me. We would be open to a PR that implements the `closeOnSave` option.

**Reply by slavanossar** [reply]
Finally got around to this! PR here #11646
